addons/im_livechat/models/discuss_channel_member.py

```diff
@@ -10,6 +10,42 @@
 class DiscussChannelMember(models.Model):
     _inherit = 'discuss.channel.member'
 
+    channel_member_history_ids = fields.One2many("im_livechat.channel.member.history", "member_id")
+    member_type = fields.Selection(
+        [("agent", "Agent"), ("visitor", "Visitor"), ("bot", "Chatbot")],
+        compute="_compute_member_type",
+        inverse="_inverse_member_type",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        members = super().create(vals_list)
+        for member in members:
+            if member.channel_id.channel_type != "livechat" or member.member_type:
+                continue
+            if member.guest_id:
+                member.member_type = "visitor"
+            elif member.partner_id.chatbot_script_ids:
+                member.member_type = "bot"
+            else:
+                member.member_type = "agent"
```

Comment:
Why do we need to guess here? It should always be provided.

Technically we can't really invite guests, and a partner could actually be a visitor, so this seems very arbitrary/too heavily relying on some flows.